### PR TITLE
Added commutativity in String/Bytes multiplication

### LINKED ISSUE #17506: Should string/bytes multiplication be commutative?
Today we can do:

```chapel
writeln("foo"*3);
writeln(b"foo"*3);
```

but we cannot do

```chapel
writeln(3*"foo");
writeln(3*b"foo");  // I was curious whether this would parse, it does (but doesn't resolve)
```

because we don't have the overloads of these operator with symmetric arguments.

I think we should support `int*string` and `int*bytes` operations, as well.

### PR DESCRIPTION
**Pull request by shubhamkmr04**
I have added the feature of `int*string` and `int*bytes` multiplication and added some documentation in `Bytes.html` while explaining the example.
Resolve #17506

### CODE REVIEW

**Comment by shubhamkmr04** [comment]
@e-kayrakli Please review.

File: test/types/bytes/multiplication.chpl [hunk 1]
```diff
@@ -0,0 +1,7 @@
+writeln(b"Hello! " * 3);
+writeln(3 * b"Hello! ");
+
+writeln(b"" * 3);
+writeln(3 * b"");
+
+writeln(b"" * 0);
```
Comment (e-kayrakli) [inline test/types/bytes/multiplication.chpl:7]: Could you also test a non-empty string multiplied by 0? I see that you have it for the bytes type but not for this one.

**Review by e-kayrakli — COMMENTED** [review]
I just asked one more case to be tested. And while making the test change, could you also change it in a way that it prints something before printing the resulting string? This looks like the good file has bunch of extra lines at the end (though they show that the operator returns empty string). What I am saying is basically:

```chapel
writeln("Result: ", 3*"foo");
```

File: test/types/bytes/multiplication.chpl (hunk 1 above)
Comment (shubhamkmr04) [reply test/types/bytes/multiplication.chpl:7]: @e-kayrakli I Improved my tests, Please have a look.
`test/types/string/multiplication.chpl`
>writeln("Hello !" * 5);
writeln(5 * "Hello !");
writeln("Result:" , "" * 4);
writeln("Result:" , 4 * "");
writeln("Result:" , "" * 0);
writeln("Result:" , 0 * "");
writeln("Result:" , "Hello !" * 0);
writeln("Result:" , 0 * "Hello !");
 
`test/types/bytes/multiplication.chpl`
>writeln(b"Hello! " * 3);
writeln(3 * b"Hello! ");
writeln("Result:" , b"" * 3);
writeln("Result:" , 3 * b"");
writeln("Result:" , b"" * 0);
writeln("Result:" , 0 * b"");
writeln("Result:" , b"Hello!" * 0);
writeln("Result:" , 0 * b"Hello!");

**Comment by e-kayrakli** [comment]
Thanks! I don't have any further comments at the moment. I would like to wait on testing this until https://github.com/chapel-lang/chapel/pull/17503 is merged. I don't anticipate any conflicts or issues, but I think it is the right order of things.

**Comment by shubhamkmr04** [comment]
@e-kayrakli  This branch is having some conflicts, I will close this PR, I have reopened the PR here #17555

**Pull request closed without merging**